metagraph/src/graph/representation/succinct/boss.hpp

```diff
@@ -721,6 +727,29 @@ BOSS::index_range(RandomAccessIt begin, RandomAccessIt end) const {
     return std::make_tuple(succ_last(rl), ru, it);
 }
 
+std::tuple<BOSS::edge_index, BOSS::edge_index, size_t>
+BOSS::index_range(std::string_view str) const {
+    assert(str.size() <= get_k());
+
+    auto encoded = encode(str);
+
+    // nothing to call if the suffix contains invalid characters
+    if (std::find(encoded.begin(), encoded.end(), alph_size) != encoded.end())
+        return {};
+
+    auto range = index_range(encoded.begin(),
+                             std::min(encoded.begin() + get_k(), encoded.end()));
+
+    if (std::get<0>(range) == 0 || std::get<1>(range) == 0)
+        return {};
+
+    if (std::get<2>(range) == encoded.begin()) {
+        return {};
+    }
+
+    return { std::get<0>(range), std::get<1>(range), std::get<2>(range) - encoded.begin() };
+}
```

Comment:
better do this in DBGSuccinct. For example, you decode the sequence here, which is useful in line 289 in dbg_succinct.cpp

Reply:
I checked again. It's used in two places. But still, let's not add this.
All these checks can be compactly rewritten there, and, in fact, you don't need all of them.
For instance, instead of checking (std::get<2>(range) != encoded.begin()), you can just check if the range is non-zero.